build.go

```diff
@@ -628,16 +642,23 @@ func randString(n int) string {
 	return string(b)
 }
 
-func processVolumes(volumes []string) (processed []string, warnings []string, err error) {
-	// Assume a linux container
-	parser := mounts.NewParser(mounts.OSLinux)
+func processVolumes(imgOS string, volumes []string) (processed []string, warnings []string, err error) {
+	parserOS := mounts.OSLinux
+	if imgOS == "windows" {
+		parserOS = mounts.OSWindows
+	}
+	parser := mounts.NewParser(parserOS)
 	for _, v := range volumes {
 		volume, err := parser.ParseMountRaw(v, "")
 		if err != nil {
 			return nil, nil, errors.Wrapf(err, "platform volume %q has invalid format", v)
 		}
 
-		for _, p := range [...]string{"/cnb", "/layers"} {
+		sensitiveDirs := []string{"/cnb", "/layers"}
+		if imgOS == "windows" {
+			sensitiveDirs = []string{`c:/cnb`, `c:\cnb`, `c:/layers`, `c:\layers`}
+		}
+		for _, p := range sensitiveDirs {
 			if strings.HasPrefix(volume.Spec.Target, p) {
```

Comment:
Is it worth lowercasing the target, in case the user passes `C:\`? Unless `volume.Spec.Target` sanitizes this...

Reply:
Good catch! Will do